doc/source/devel/writing-docs.rst

```diff
@@ -0,0 +1,133 @@
+
+The ``par`` extension
+=====================
+
+We have also written an extension for Sphinx to assist with formatting Cello/Enzo-E's parameters, called ``par``.
+At present, this extension provides 3 primary features:
+
+    1. A directive called, :rst:dir:`par:parameter`, that is to be used when defining a new parameter in the reference section.
+       Parameters defined with this directive automatically provide an anchor point to facillitate cross-referencing (i.e. using the :rst:role:`par:param` role).
+
+    2. Pretty-formatting of parameter names.
+       This is done by the :rst:dir:`par:parameter` directive, as well as the :rst:role:`par:param` and :rst:role:`par:paramfmt` roles.
+       Note, this functionallity supports special formatting of components of parameter names enclosed by angle brackets (e.g. ``:par:paramfmt:`Boundary:<condition>:type``` and ``:par:paramfmt:`Initial:value:<field>``` are respectively rendered as :par:paramfmt:`Boundary:<condition>:type` and :par:paramfmt:`Initial:value:<field>`)
+
+    3. Pretty-formatting of parameter types with the :rst:role:`par:typefmt` role.
+       For example, ``:par:typefmt:`list ( float )``` gets rendered as :par:typefmt:`list ( float )`.
+
+
+
+Defining a Parameter
+--------------------
+
+When defining a new parameter in the reference section, you should use the following directive:
+
+.. rst:directive:: par:parameter
+
+   This directive should be used when defining a new Cello/Enzo-E parameter.
+
+   The argument passed to the directive is the name of the parameter being documented.
+   After the line ``.. par:parameter:: ParamName``, you should leave a blank line, and then you should define a list of fields (e.g. ``Summary``, ``Type``, ``Default``, ``Scope``) that provide an overview of the parameter.
+   Finally, you generally provide an extended description after the field-list.
+
+   .. rubric:: Example
+
+   It is insightful to consider a concrete example.
+   Below we show how to do this for the :par:paramfmt:`Method:heat:alpha` parameter:
+
+   .. code-block:: rst
+
+       .. par:parameter:: Method:heat:alpha
+
+          :Summary:    :s:`Parameter for the forward euler heat equation solver`
+          :Type:       :par:typefmt:`float`
+          :Default:    :d:`1.0`
+          :Scope:      :z:`Enzo`
+
+          :e:`Thermal diffusivity parameter for the heat equation.`
+
+   This snippet will be rendered as the following (cross-referencing to this output has been disabled):
+
+   .. par:parameter:: Method:heat:alpha
+      :noindex:
+
+      :Summary:    :s:`Parameter for the forward euler heat equation solver`
+      :Type:       :par:typefmt:`float`
+      :Default:    :d:`1.0`
+      :Scope:      :z:`Enzo`
+
+      :e:`Thermal diffusivity parameter for the heat equation.`
+
+   .. rubric:: Additional Notes
+
+   This directive also accepts the ``:noindex:`` option, but you shouldn't worry about that (it's only used for writing this example).
```

Comment:
I suggest moving the stuff about :noindex: to a comment in the rst file, since it's really only relevant to people editing the source and wondering what it's doing there.

Reply:
Good idea